Lib/idlelib/run.py

```diff
@@ -271,7 +271,13 @@ def print_exc(typ, exc, tb):
                        "debugger_r.py", "bdb.py")
             cleanup_traceback(tbe, exclude)
             traceback.print_list(tbe, file=efile)
-        lines = get_message_lines(typ, exc, tb)
+        if not isinstance(exc, NameError) and not isinstance(exc, AttributeError):
+            lines = get_message_lines(typ, exc, tb)
+        else:
+            if "\n" in str(exc):
+                lines = [f"{typ.__name__}: {str(exc)}"]
+            else:
+                lines = get_message_lines(typ, exc, tb)
```

Comment:
An even better change if it works.
1.  Remove 'tb' from the signature of get_message_lines here
```suggestion
        lines = get_message_lines(typ, exc)
```
and its definition above on line 231 and at least 1 one call in idle_test/test_run.py
2. Replace `tb` with `None` in the call to `sys.__excepthook__`.  I tested this and the result is to not add any traceback lines.
3. On the next line, `return [err.getvalue() + "\n"]`.  There should no longer be traceback lines to remove.  err.getvalue() should be a single string with an embedded newline, to which we add a final newline as before.